### PR TITLE
Added and documented missing modules for QoSHandler, SizeLimitHandler and ThreadLimitHandler.

### PR DESCRIPTION
**Pull request by sbordet**
Moved SizeLimitHandler from oej.server to oej.server.handler.

### CODE REVIEW

File: documentation/jetty/modules/programming-guide/pages/server/http.adoc [hunk 1]
```diff
@@ -814,6 +814,29 @@ include::code:example$src/main/java/org/eclipse/jetty/docs/programming/server/ht
...
 
+[[handler-use-conditional]]
+==== ConditionalHandler
+
+`ConditionalHandler` is an abstract `Handler` that matches conditions about the request, and allows subclasses to handle the request differently depending on whether the conditions have been met or not.
+
+Subclasses must override methods:
+
+* `boolean onConditionsMet(Request, Response, Callback)`, to handle the request when the conditions are met.
+* `boolean onConditionsNotMet(Request, Response, Callback)`, to handle the request when the conditions are not met.
```
Comment (gregw) [inline documentation/jetty/modules/programming-guide/pages/server/http.adoc]: You should mention the sub types:
 + ElseNext
 + SkipNext
 + DontHandle
 + Reject

which implement one or both of these methods with common behaviours

File: documentation/jetty/modules/programming-guide/pages/server/http.adoc [hunk 2]
```diff
@@ -1040,6 +1063,8 @@ Refer to the `EventsHandler` link:{javadoc-url}/org/eclipse/jetty/server/handler
 
 `QoSHandler` allows web applications to limit the number of concurrent requests, therefore implementing a quality of service (QoS) mechanism for end users.
```
Comment (gregw) [inline documentation/jetty/modules/programming-guide/pages/server/http.adoc]: Is it really limiting "concurrent requests" ?  It does not limit the number of request, just the number that are actively being handled. It is limiting threads running concurrent requests or something to that effect.

We should also say why this is a good thing

File: jetty-core/jetty-server/src/main/config/modules/size-limit.mod [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+[depends]
+server
+
+[xml]
+etc/jetty-size-limit.xml
+
+[ini-template]
+#tag::documentation[]
+## The maximum request content size, in bytes.
```
Comment (gregw) [inline jetty-core/jetty-server/src/main/config/modules/size-limit.mod]: define -1

File: jetty-core/jetty-server/src/main/config/modules/size-limit.mod [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+# jetty.sizeLimit.maxRequestContentSize=-1
+
+## The maximum response content size, in bytes.
```
Comment (gregw) [inline jetty-core/jetty-server/src/main/config/modules/size-limit.mod]: define -1

File: jetty-core/jetty-server/src/main/config/etc/jetty-qos.xml [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+          <Call class="java.time.Duration" name="ofMillis">
+            <Arg type="long">
+              <Property name="jetty.qos.maxSuspendDuration" default="0" />
+            </Arg>
+          </Call>
+        </Set>
+      </New>
+    </Arg>
+  </Call>
+</Configure>
```
Comment (gregw) [inline jetty-core/jetty-server/src/main/config/etc/jetty-qos.xml:62]: Should we do the same as jetty-dos.xml for all ConditionalHandlers and includes something like:
```xml

        <Call name="includeInetAddressPattern">
          <Arg>
            <Call class="org.eclipse.jetty.util.StringUtil" name="csvSplit">
              <Arg><Property name="jetty.qos.include.inet" default="" /></Arg>
            </Call>
          </Arg>
        </Call>
        <Call name="excludeInetAddressPattern">
          <Arg>
            <Call class="org.eclipse.jetty.util.StringUtil" name="csvSplit">
              <Arg><Property name="jetty.qos.exclude.inet" default="" /></Arg>
            </Call>
          </Arg>
        </Call>
        <Call name="includePath">
          <Arg>
            <Call class="org.eclipse.jetty.util.StringUtil" name="csvSplit">
              <Arg><Property name="jetty.qos.include.path" default="" /></Arg>
            </Call>
          </Arg>
        </Call>
        <Call name="excludePath">
          <Arg>
            <Call class="org.eclipse.jetty.util.StringUtil" name="csvSplit">
              <Arg><Property name="jetty.qos.exclude.path" default="" /></Arg>
            </Call>
          </Arg>
        </Call>
```

**Review by gregw — CHANGES_REQUESTED** [review]

File: documentation/jetty/modules/programming-guide/pages/server/http.adoc (hunk 1 above)
Comment (sbordet) [reply documentation/jetty/modules/programming-guide/pages/server/http.adoc]: Done.

File: jetty-core/jetty-server/src/main/config/etc/jetty-qos.xml (hunk 5 above)
Comment (sbordet) [reply jetty-core/jetty-server/src/main/config/etc/jetty-qos.xml:62]: Done, added also `[in|ex]cludeMethod`, and also for `jetty-thread-limit.xml`.

File: documentation/jetty/modules/programming-guide/pages/server/http.adoc (hunk 2 above)
Comment (sbordet) [reply documentation/jetty/modules/programming-guide/pages/server/http.adoc]: Done.

File: jetty-core/jetty-server/src/main/config/modules/size-limit.mod (hunk 3 above)
Comment (sbordet) [reply jetty-core/jetty-server/src/main/config/modules/size-limit.mod]: Done.

File: jetty-core/jetty-server/src/main/config/modules/size-limit.mod (hunk 4 above)
Comment (sbordet) [reply jetty-core/jetty-server/src/main/config/modules/size-limit.mod]: Done.

File: jetty-core/jetty-server/src/main/config/etc/jetty-qos.xml [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+          <Arg>
+            <Call class="org.eclipse.jetty.util.StringUtil" name="csvSplit">
+              <Arg><Property name="jetty.qos.include.inet" default="" /></Arg>
+            </Call>
+          </Arg>
+        </Call>
+        <Call name="excludeInetAddressPattern">
+          <Arg>
+            <Call class="org.eclipse.jetty.util.StringUtil" name="csvSplit">
+              <Arg><Property name="jetty.qos.exclude.inet" default="" /></Arg>
```
Comment (gregw) [inline jetty-core/jetty-server/src/main/config/etc/jetty-qos.xml:55]: Actually, we should have done this for the eager-content XML in the PR that I just merged.....
So can you pull, update and add this to that xml as well.

**Review by gregw — CHANGES_REQUESTED** [review]
All good, but can you also fix the eager xml that I just merged?

File: jetty-core/jetty-server/src/main/config/etc/jetty-qos.xml (hunk 6 above)
Comment (sbordet) [reply jetty-core/jetty-server/src/main/config/etc/jetty-qos.xml:55]: @gregw this PR is for 12.0.x.
Please approve it and I'll do what you ask above in the merge to 12.1.x, sounds good?

**Review by gregw — APPROVED** [review]

**Pull request merged**